File: Real-estate-market-IB190099.Services/Database/DemandAdvertise.cs
using System;
using System.Collections.Generic;

namespace Real_estate_market_IB190099.Services.Database;

public partial class DemandAdvertise
{
    public int Id { get; set; }

    public string? Status { get; set; }

    public string? Type { get; set; }

    public string? Description { get; set; }

    public double? MinQuadrature { get; set; }

    public double? MaxQuadrature { get; set; }

    public string? Location { get; set; }

    public int? Floors { get; set; }

    public string? PropertyType { get; set; }

    public int? Rooms { get; set; }

    public byte? Parking { get; set; }

    public byte? Water { get; set; }

    public byte? Electricity { get; set; }

    public int? UserId { get; set; }

    public DateTime? DateCreated { get; set; }

    public virtual ICollection<DemandMessage> DemandMessages { get; set; } = new List<DemandMessage>();

    public virtual ICollection<SavedDemandAdvertise> SavedDemandAdvertises { get; set; } = new List<SavedDemandAdvertise>();

    public virtual User? User { get; set; }
}

File: Real-estate-market-IB190099.mobile/realestatemobile/lib/model/demand_advertise.dart
// ignore_for_file: non_constant_identifier_names

import 'dart:core';
import 'package:json_annotation/json_annotation.dart';
import 'package:realestatemobile/model/user.dart';

part 'demand_advertise.g.dart';

@JsonSerializable()
class DemandAdvertise {
  DemandAdvertise() {}
  int? id;
  String? status;
  String? type;
  String? description;
  double? minQuadrature;
  String? location;
  double? maxQuadrature;
  int? floors;
  String? propertyType;
  int? rooms;
  int? parking;
  int? water;
  int? electricity;
  int? userId;
  DateTime? dateCreated;
  User? user;

  factory DemandAdvertise.fromJson(Map<String, dynamic> json) =>
      _$DemandAdvertiseFromJson(json);

  /// Connect the generated [_$DemandAdvertiseToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$DemandAdvertiseToJson(this);
}
